tuplex/codegen/src/FunctionRegistry.cc

```diff
@@ -966,6 +966,301 @@ namespace tuplex {
             return SerializableValue(resVal, resSize);
         }
 
+        codegen::SerializableValue FunctionRegistry::createMathIsNanCall(llvm::IRBuilder<>& builder, const python::Type &argsType,
+                                                     const python::Type &retType,
+                                                     const std::vector<tuplex::codegen::SerializableValue> &args) {
+            using namespace llvm;
+            auto& context = builder.GetInsertBlock()->getContext();
+            auto val = args.front();
+            auto type = argsType.parameters().front();
+            
+            if (python::Type::F64 == type) {
+                llvm::Value* i64Val = builder.CreateBitCast(val.val, llvm::Type::getInt64Ty(context));
+                _env.printValue(builder, val.val, "double/python float value\n");
+                
+                auto shiftedVal = builder.CreateLShr(i64Val, 32);
```

Comment:
I can not really understand what is going on in the following instructions. Please add comments on what is being generated here, specifically what the magic numbers like `2147483647` are and where they are coming from.

Reply:
this has been addressed in the comments. Why not use hex notation though?